src/RevProc.cc

```diff
@@ -956,309 +956,308 @@ RevInst RevProc::DecodeCompressed(uint32_t Inst){
 }
 
 RevInst RevProc::DecodeRInst(uint32_t Inst, unsigned Entry){
-  RevInst DInst;
+  RevInst RInst;
 
   // cost
   RegFile[threadToDecode].cost  = InstTable[Entry].cost;
 
   // encodings
-  DInst.opcode  = InstTable[Entry].opcode;
-  DInst.funct3  = InstTable[Entry].funct3;
-  DInst.funct2  = 0x0;
-  DInst.funct7  = InstTable[Entry].funct7;
+  RInst.opcode  = InstTable[Entry].opcode;
+  RInst.funct3  = InstTable[Entry].funct3;
+  RInst.funct2  = 0x0;
+  RInst.funct7  = InstTable[Entry].funct7;
 
   // registers
-  DInst.rd      = 0x0;
-  DInst.rs1     = 0x0;
-  DInst.rs2     = 0x0;
-  DInst.rs3     = 0x0;
+  RInst.rd      = 0x0;
+  RInst.rs1     = 0x0;
+  RInst.rs2     = 0x0;
+  RInst.rs3     = 0x0;
 
   if( InstTable[Entry].rdClass != RegUNKNOWN ){
-    DInst.rd  = DECODE_RD(Inst);
+    RInst.rd  = DECODE_RD(Inst);
   }
   if( InstTable[Entry].rs1Class != RegUNKNOWN ){
-    DInst.rs1  = DECODE_RS1(Inst);
+    RInst.rs1  = DECODE_RS1(Inst);
   }
   if( InstTable[Entry].rs2Class != RegUNKNOWN ){
-    DInst.rs2  = DECODE_RS2(Inst);
+    RInst.rs2  = DECODE_RS2(Inst);
   }
 
   // imm
-  DInst.imm     = 0x0;
+  RInst.imm     = 0x0;
 
   // SP/DP Float
-  DInst.fmt     = 0;
-  DInst.rm      = 0;
+  RInst.fmt     = 0;
+  RInst.rm      = 0;
 
   // Size
-  DInst.instSize  = 4;
+  RInst.instSize  = 4;
 
   // Decode the atomic RL/AQ fields
-  if( DInst.opcode == 0b0101111 ){
-    DInst.rl = DECODE_RL(Inst);
-    DInst.aq = DECODE_AQ(Inst);
+  if( RInst.opcode == 0b0101111 ){
+    RInst.rl = DECODE_RL(Inst);
+    RInst.aq = DECODE_AQ(Inst);
   }
 
   // Decode any ancillary SP/DP float options
   if( IsFloat(Entry) ){
-    DInst.rm = DECODE_FUNCT3(Inst);
+    RInst.rm = DECODE_FUNCT3(Inst);
   }
 
-  DInst.compressed = false;
+  RInst.compressed = false;
 
-  return DInst;
+  return RInst;
 }
 
 RevInst RevProc::DecodeIInst(uint32_t Inst, unsigned Entry){
-  RevInst DInst;
+  RevInst IInst;
 
   // cost
   RegFile[threadToDecode].cost  = InstTable[Entry].cost;
 
   // encodings
-  DInst.opcode  = InstTable[Entry].opcode;
-  DInst.funct3  = InstTable[Entry].funct3;
-  DInst.funct2  = 0x0;
-  DInst.funct7  = 0x0;
+  IInst.opcode  = InstTable[Entry].opcode;
+  IInst.funct3  = InstTable[Entry].funct3;
+  IInst.funct2  = 0x0;
+  IInst.funct7  = InstTable[Entry].funct7; // For ecall, ebreak, slli, srli, srai
```

Comment:
this is wrong

Reply:
> this is wrong

Setting funct 7 to 0 will not be able to differentiate `srli` from `srai`.